backend/test/capsule.unit.test.ts

```diff
@@ -0,0 +1,71 @@
+import { BigNumber, Contract, ContractFactory } from "ethers";
+import { expect, assert } from "chai";
+const { ethers, network } = require("hardhat");
+
+describe("capsule unit tests", function () {
+    const amount1 = ethers.utils.parseEther("1000");
+    let capsuleFactory: Contract,
+        erc20Capsule: Contract,
+        nftCapsule: Contract,
+        mixedCapsule: Contract,
+        wethToken: Contract,
+        usdcToken: Contract,
+        wbtcToken: Contract,
+        daiToken: Contract,
+        user: Contract,
+        user2: Contract;
+
+    beforeEach(async function () {
+        const accounts = await ethers.getSigners(2);
+        user = accounts[0];
+        user2 = accounts[1];
+
+        const wethTokenFactory: ContractFactory = await ethers.getContractFactory("WETH");
+        wethToken = await wethTokenFactory.deploy();
+        // prettier-ignore
+        await wethToken.deployed();
+
+        const daiTokenFactory: ContractFactory = await ethers.getContractFactory("DAI");
+        daiToken = await daiTokenFactory.deploy();
+        // prettier-ignore
+        await daiToken.deployed();
+
+        const usdcTokenFactory: ContractFactory = await ethers.getContractFactory("USDC");
+        usdcToken = await usdcTokenFactory.deploy();
+        // prettier-ignore
+        await usdcToken.deployed();
+
+        const wbtcTokenFactory: ContractFactory = await ethers.getContractFactory("WBTC");
+        wbtcToken = await wbtcTokenFactory.deploy();
+        // prettier-ignore
+        await wbtcToken.deployed();
```

Comment:
Need rewrite, due to the GenericERC20.sol introduction you have to specify all 3 values at the constructor and also it might be good to deploy a fixture where your deployments will run before your test